src/lib/components/ResourceTypeField.js

```diff
@@ -1,81 +1,114 @@
 // This file is part of React-Invenio-Deposit
-// Copyright (C) 2020 CERN.
-// Copyright (C) 2020 Northwestern University.
+// Copyright (C) 2020-2021 CERN.
+// Copyright (C) 2020-2021 Northwestern University.
 //
 // React-Invenio-Deposit is free software; you can redistribute it and/or modify it
 // under the terms of the MIT License; see LICENSE file for more details.
 
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { Field } from 'formik';
+import { FastField } from 'formik';
 import _get from 'lodash/get';
+import { Form } from 'semantic-ui-react';
 
-import { FieldLabel, GroupField, SelectField } from 'react-invenio-forms';
+import { FieldLabel } from 'react-invenio-forms';
 
 export class ResourceTypeField extends Component {
-  groupErrors = (errors) => {
-    for (const field in errors) {
-      if (field.startsWith(this.props.fieldPath)) {
-        return { content: _get(errors, this.props.fieldPath) };
-      }
+  groupErrors = (errors, fieldPath) => {
+    const fieldErrors = _get(errors, fieldPath);
+    if (fieldErrors) {
+      return { content: fieldErrors };
     }
     return null;
   };
 
-  renderResourceTypeField = (formikBag) => {
-    const typeFieldPath = `${this.props.fieldPath}.type`;
-    const subtypeFieldPath = `${this.props.fieldPath}.subtype`;
+  createOptions = (propsOptions) => {
+    let options = [];
+    for (const type of propsOptions.type) {
+      const subtypes = propsOptions.subtype.filter(
+        (e) => e['parent-value'] === type.value
+      );
+
+      if (subtypes.length > 0) {
+        // Push options corresponding to each subtype
+        options.push(
+          ...subtypes.map((subtype) => ({
+            // NOTE: In this version of semantic-ui-react (0.88.2),
+            // using icon key doesn't show icon in selected text. This does:
+            text: <><i className={type.icon + " icon"}></i><span className="text">{subtype["parent-text"] + " / " + subtype.text}</span></>,
+            value: subtype["parent-value"] + "/" + subtype.value,
+            values: {type: type.value, subtype: subtype.value}
+          }))
+        );
+      } else {
+        // Push an option corresponding to the type only
+        options.push({
+            icon: type.icon,
+            text: type.text,
+            value: type.value,
+            values: {type: type.value}
```

Comment:
Just out of curiosity...the dropdown options can also include arbitrary extra data?

Reply:
Yep, they are just objects. When the user clicks on the option, the `data` object that the click handler receives has `value` only set to the `value` of the option, but we can get back the associated other data by using that `value` as a unique key in the `options` array.